app/admin/tenants/onboarding.py

```diff
@@ -126,94 +124,129 @@ async def onboard_issuer(
+
         await acapy_ledger.register_nym_on_ledger(
-            endorser_controller,
-            did=issuer_did.did,
-            verkey=issuer_did.verkey,
-            alias=name,
+            endorser_controller, did=issuer_did.did, verkey=issuer_did.verkey
         )
+
         await acapy_ledger.accept_taa_if_required(issuer_controller)
-        await acapy_wallet.set_public_did(issuer_controller, issuer_did.did)
-
-    endorser_did = await acapy_wallet.get_public_did(controller=endorser_controller)
-
-    # Make sure the issuer has a connection with the endorser
-    invitation = await endorser_controller.out_of_band.create_invitation(
-        auto_accept=True,
-        body=InvitationCreateRequest(
-            handshake_protocols=["https://didcomm.org/didexchange/1.0"],
-            use_public_did=True,
-        ),
-    )
-
-    logger.info(
-        f"Starting webhook listener for connections with wallet id {issuer_wallet_id}"
-    )
-
-    wait_for_event, _ = await start_listener(
-        topic="connections", wallet_id=issuer_wallet_id
-    )
-    endorser_wait_for_event, _ = await start_listener(
-        topic="connections", wallet_id="admin"
-    )
-
-    logger.debug("Receiving connection invitation")
-
-    # FIXME: make sure the connection with this alias doesn't exist yet
-    # Or does use_existing_connection take care of this?
-    connection_record = await issuer_controller.out_of_band.receive_invitation(
-        auto_accept=True,
-        use_existing_connection=True,
-        body=invitation.invitation,
-        alias=ACAPY_ENDORSER_ALIAS,
-    )
-
-    logger.debug(
-        f"Waiting for connection with id {connection_record.connection_id} to be completed"
-    )
-
-    # Wait for connection to be completed before continuing
-    try:
-        await wait_for_event(
-            filter_map={
-                "connection_id": connection_record.connection_id,
-                "state": "completed",
-            }
+        # TODO: This doesn't need endorsement as of 0.7.5-rc0 onward - bug in 0.7.4
+        # Change this in future versions
+        await acapy_wallet.set_public_did(
+            issuer_controller,
+            did=issuer_did.did,
+            connection_id=connection_record.connection_id,
+            create_transaction_for_endorser=True,
+            is_endorsed=True,
         )
 
-        endorser_connection = await endorser_wait_for_event(
-            filter_map={
-                "invitation_msg_id": invitation.invi_msg_id,
-                "state": "completed",
-            }
+        try:
+            await endorser_wait_for_transaction(
+                filter_map={
+                    "state": "request-received",
+                }
+            )
+        except TimeoutError:
+            raise CloudApiException("Error creating connection with endorser", 500)
+
+        txns = await endorser_controller.endorse_transaction.get_records()
```

Comment:
As per my previous comment, I don't think we need to endorse the transaction. However if we do need it, I think we could take the transaction_id from the `endorser_wait_for_transaction` return value event, and that would save us to retrieve _all_ endorsement requests (which wouldn't really scale)

Reply:
Let me try this again. I think your suggestion works in 0.7.5-rc0 onwards (and current main) but not in 0.7.4 - but will give it a try

Reply:
thx for the hint here. extracting it from `endorser_wait_for_transaction` is the better way